flatten-array.md

```diff
@@ -0,0 +1,19 @@
+The challenge is to write a function that accepts an arbitrarily-deep nested Array-like structure and returns a flattened structure with any null values removed without using the reduce method.
+
+For Example
+
+flattener.flatten([0, 2, [[2, 3], 8, 100, null, [[null]]], -2]);
+
+returns [0, 2, 2, 3, 8, 100, -2]
+
+
+flattener.clear()
+
+returns []
+
+Tips
+
+* Define clear first before the function flatten
+* Think about closures and know what you have access to within a function
```

Comment:
Not all languages have closures. If these tips are language or implementation-specific, they could be added to a `HINTS.md` file inside the exercise directory alongside the test suite and the reference solution. This would get included into the README for just that language track.